### ISSUE TITLE
Add a duplicate/copy button to inline panels

### ISSUE THREAD

**Issue by sighmon** [wagtail/wagtail#7711]
### Is your proposal related to a problem?

Our staff is frustrated adding lots of culsterable/orderable inline panels with similar fields.

### Describe the solution you'd like

A Duplicate button, similar to the [StreamField Blocks](https://github.com/wagtail/wagtail/search?q=DuplicateButton) duplicate button, next to the `up`, `down`, and `delete` buttons.

![inline-panel-duplicate-button](https://user-images.githubusercontent.com/314298/141927173-eea47b26-fdfd-4dcd-ab56-b44a4330029d.png)

### Describe alternatives you've considered

None.

### Additional context

We have an `Occurrence` class that inherits from `BaseOccurrence` below. Our staff want to be able to copy the inline panel for an occurrence for a different date, without having to manually copy/paste the venue/ticketing_id/purchase_url/status.

```Python
class BaseOccurrence(ClusterableModel, Orderable):
    """Abstract base class for Occurrences."""

    datetime = models.DateTimeField()
    venue = models.CharField(max_length=255, blank=True)
    ticketing_id = models.IntegerField(null=True, blank=True, verbose_name="Ticketing ID")
    purchase_url = models.URLField(blank=True, verbose_name="Purchase URL")
    status = models.CharField(max_length=255, blank=True)

    panels = [
        FieldPanel("datetime"),
        FieldPanel("ticketing_id"),
        FieldPanel("venue"),
        FieldPanel("event"),
        FieldPanel("status"),
        FieldPanel("purchase_url"),
    ]

    api_fields = [
        APIField("datetime"),
        APIField("ticketing_id"),
        APIField("venue"),
        APIField("event"),
        APIField("status"),
        APIField("purchase_url"),
    ]

    class Meta:
        abstract = True
        ordering = ["datetime"]
```

**Comment by lb-** [comment]
Rough version of the InlinePanel JS ported to Stimulus JS with copy and undo delete functionality.
https://github.com/lb-/stimulus-starter/blob/main/src/controllers/inline-panel-controller.js
As of this comment - nested inline panels copying are not functional but this could be a step in the right direction.
Depending on resolution of lightweight framework we could adopt this as a start or port the JS back into something else we end up using.

https://github.com/lb-/stimulus-starter/blob/main/src/controllers/inline-panel-controller.js#L96

**Comment by gasman** [comment]
This is really interesting, and also a bit of a bombshell :-)

The StreamField Telepath rewrite was pretty much based on the premise that we can't possibly populate or duplicate arbitrary Django form widgets from client-side code without having specific field-by-field knowledge of what those widgets are - and it looks like that's essentially what you're aiming to do here.

If it can be made to work reliably, then that's great news - but it also has interesting implications for the future of Telepath :-) On the flipside, if it works on (say) 90% of setups, then that's still going to make it a bit of a hard sell to switch from this to a Telepath-based implementation in future, since that 90% may well include widgets that haven't been hooked up to work with Telepath yet - and so from an end-user perspective it would look like we're breaking things at the same time as fixing others.

**Comment by lb-** [comment]
@gasman yeah - if you have not already - I'd love to get your thoughts into he research done for the lightweight framework approach. 
https://github.com/wagtail/wagtail/discussions/7689#discussioncomment-2037913

But the tldr is - Stimulus gives us a simple abstraction over targeting elements, registering event listeners and maintaining simple state (as DOM attributes). Any behaviour linked to an element via its controller gets automatically instantiated by simply adding it to the DOM (uses MutationObserver under the hood). 

This means we can get to a pretty solid coverage of all existing JS / jQuery implementations without ever having to call init. 

I've thought a lot about where Telepah fits in and I've covered it slightly in the investigation document link above but will try to clarify some ideas over the weekend. But I think we would approach a new component like a. CSS and HTML first, b. add stimulus for behaviour as needed, c. add telepath for the initial data for more complex python driven classes on the clientside and d. Use React for the really complex parts (which should be minimal). 

In all cases, aside from CSS - Stimulus can still be the init entry. Just the data source may be located in another script or maybe API response.
